### ISSUE TITLE
SCM View Should Allow Multi-Select

### ISSUE THREAD

**Issue by tsmaeder** [eclipse-theia/theia#7659]
One can only select single entries in the "Source Control" view. This is annoying when trying to stage multiple changes in git, for example.

To reproduce:

1. Make changes in multiple files in a git project in your workspace
2. Go to the "Source Control" view
3. Observe: you see the changed files under "Changes"
4. Shift-Click or Ctrl-Click multiple files in the view
5. Observe: as soon as you click a second file, the selection on the first goes away

Multi-select works in this view in VS Code (and it's mighty useful, I dare say)

**Comment by vince-fugnitto** [comment]
@tsmaeder is it fine to close the original, the issue is a duplicate of https://github.com/eclipse-theia/theia/issues/1318

**Comment by tsmaeder** [comment]
🤷 I was looking for "scm" issues. I would suspect this is not git specific?

**Comment by vince-fugnitto** [comment]
> 🤷 I was looking for "scm" issues. I would suspect this is not git specific?

I believe it is the same problem, the original was named git since we had no notion of `scm` before. I'll close the original in favor of this one as it is newer and more up to date :)

**Comment by westbury** [comment]
This should be easy once https://github.com/eclipse-theia/theia/pull/7505 is merged.  That PR re-writes the source control view to use tree-widget, so we can use the same selection code.  Also that PR adds support for an array of ScmResource in the context menu args (previously just a single ScmResource allowed), and downstream dugite API adjusted to accept multiples.  This was actually done to support actions on folders.  As I already know the changes required here, I'll prepare the PR and submit it once https://github.com/eclipse-theia/theia/pull/7505 is merged.
